examples/angular/src/app/guards/can-match-page.guard.ts

```diff
@@ -0,0 +1,42 @@
+import { inject } from '@angular/core';
+import { CanMatchFn, Route, Router, UrlSegment } from '@angular/router';
+
+import { DOTCMS_CLIENT_TOKEN } from '../client-token/dotcms-client';
+import { DotCMSPageAsset } from '@dotcms/angular';
+
+export const canMatchPage: CanMatchFn = async (
+  route: Route,
+  segments: UrlSegment[]
+) => {
+  const router = inject(Router);
+  const client = inject(DOTCMS_CLIENT_TOKEN);
+
+  const path = segments?.map((segment) => segment.path)?.join('/') || '/';
+  const initialUrl = router.getCurrentNavigation()?.initialUrl;
+  const queryParams = initialUrl?.queryParams || {};
+
+  const { mode, variantName, language_id } = queryParams;
```

Comment:
I think this can lead to an error if the object is empty.

Reply:
If you look above, you can see that I'm using `const queryParams = initialUrl?.queryParams || {};`.

It should never be empty.

Reply:
I mean the `{}' does not have `mode` `variantName` or `languageId` inside, there's where it could fail the destructuring

Reply:
The destructuring fails when you are trying to destructure a `non-object` Datatype

### Safe Case
![safe-case](https://github.com/dotCMS/core/assets/72418962/1ab63399-88c1-4f43-9bba-f5deae6fa730)

### Error Case
![error-case](https://github.com/dotCMS/core/assets/72418962/6283401c-1718-454a-9173-01d6a4d111c2)

However, I'll add that inside the `try/catch` statement 🙌

Reply:
Nice